inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -71,7 +74,15 @@ def arxiv_fulltext_download(obj, eng):
     )
 
     if pdf:
-        obj.log.info('PDF retrieved from arXiv for %s', arxiv_id)
+        if pdf.get_version().mimetype == 'application/pdf':
+            obj.log.info('PDF retrieved from arXiv for %s', arxiv_id)
+        else:
+            # We need to delete the failed PDF
+            del obj['files'][filename]
+            pdf.delete()
+            obj.log.warning('PDF retrieved from arXiv for %s', arxiv_id)
+            raise RuntimeError("PDF from arXiv for {arxiv_id} isn't yet ready".format(
```

Comment:
Mmh. @jacquerie using the predicate wouldn't make it any more readable because:
* I would have to always delete potential previously downloaded PDFs
* I wouldn't have to check if the last PDF is good or not. It would simply not repeat the download and leave the dangling broken data.